### PR TITLE
js: Add tests for BurnChecked instruction

### PR DESCRIPTION
**Pull request by raushan728**
Added a dedicated test suite for the `BurnChecked` instruction in `clients/js`
Changes:

- Created `clients/js/test/burnChecked.test.ts`.
- Success Tests: Verified standard burns with supply check, delegated burns, and zero-amount edge cases.
- Failure Tests: Validated errors for decimal mismatches, insufficient funds, and missing signers.

### CODE REVIEW

File: clients/js/test/burnChecked.test.ts [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+  await pipe(
+    await createDefaultTransaction(client, payer),
+    (tx) => appendTransactionMessageInstruction(burnChecked, tx),
+    (tx) => signAndSendTransaction(client, tx)
+  );
+
+  // Then the total supply should decrease to 175.
+  const { data: mintData } = await fetchMint(client.rpc, mint);
+  t.is(mintData.supply, 175n);
+});
```
Comment (joncinque) [inline clients/js/test/burnChecked.test.ts]: Rather than adding another test for this, how about just checking the supply in the first test and removing this one?
